tensorrt_llm/_torch/modules/gated_mlp.py

```diff
@@ -95,12 +96,23 @@ def __init__(self,
             [LoraModuleType.MLP_GATE_UP],
             [2 * self.intermediate_size // mapping.tp_size])
 
-    def _apply_activation(self, x):
+    def _apply_activation(self, x, *, for_lora: bool = False):
         if self.activation == F.silu:
             if self.down_proj.has_fp8_qdq:
-                return swiglu(x,
-                              quant_scale=self.down_proj.input_scale,
-                              quant_type=torch.float8_e4m3fn)
+                if for_lora:
+
+                    target = torch.bfloat16 if torch.cuda.is_bf16_supported(
+                    ) else torch.float16
+                    logger.debug(
+                        f"GatedMLP._apply_activation: LoRA path active; forcing non-FP8 activation dtype {target} (keeping activations in bf16/fp16), layer_idx={self.layer_idx}"
+                    )
+                    return swiglu(x,
+                                  quant_scale=self.down_proj.input_scale,
+                                  quant_type=target)
```

Comment:
Just realize that the quant_scale/quant_type should not be set here. The output from `gate_up_proj` should already be bf16/fp16, and set quant_scale will let the kernel multiply the scale with x, which would involve accuracy bug.

Maybe revise like this:

```suggestion
                    return swiglu(x)
```

Or change the above condition `if self.down_proj.has_fp8_qdq:` to `if self.down_proj.has_fp8_qdq and not for_lora:`

Reply:
thank you @JunyiXu-nv  for catching that! i'm not familiar with this area of the code, so missed it. This should ideally have been caught right?
I guess L0 didnt' catch it but do you know if there is any accuracy coverage in L1+ for such an accuracy bug? 

maybe i should run a post merge to make sure this PR doesn't break anything (even after the above suggestion)?